OMCompiler/Compiler/NBackEnd/Modules/2_Pre/NBEvents.mo

```diff
@@ -963,12 +991,31 @@ protected
           iter        = iter,
           eqn         = eqn_ptr,
           funcTree    = funcTree,
-          createAux   = createAux);
+          createEqn   = createEqn);
 
     eqn := match eqn
-      case Equation.ALGORITHM() algorithm
-        for stmt in eqn.alg.statements loop
-          StateEvent.fromStatement(stmt, bucket_ptr, eqn_ptr, funcTree);
+      case Equation.ALGORITHM(alg = alg) algorithm
+        new_stmts := {};
+        for stmt in alg.statements loop
+          stmt := StateEvent.fromStatement(stmt, bucket_ptr, eqn_ptr, funcTree);
+          bucket := Pointer.access(bucket_ptr);
+          new_stmts := stmt :: new_stmts;
+          if Util.isSome(bucket.aux_stmts) then
+            // add all new statements to the algorithm body
+            for tpl in Util.getOption(bucket.aux_stmts) loop
+              (cond, aux) := tpl;
+              aux               := ComponentRef.mapSubscripts(aux, function Subscript.mapExp(func = function BackendDAE.lowerComponentReferenceExp(variables = variables)));
+              new_stmt          := Statement.makeAssignment(Expression.fromCref(aux), cond.exp, ComponentRef.getSubscriptedType(aux), DAE.emptyElementSource);
+              new_stmts         := new_stmt :: new_stmts;
+            end for;
+            // remove the current statements because they have been added
+            bucket.aux_stmts   := NONE();
+            Pointer.update(bucket_ptr, bucket);
+            // save all the new stuff in our algorithm
+            alg.statements  := new_stmts;
+            eqn.alg         := Algorithm.setInputsOutputs(alg);
```

Comment:
This may be unnecessary work. We can just add `aux` to the outputs of `alg` and increase the size by `size(aux)` or something like that. The rest of `alg` should be unharmed by the new statements.

If we do that inside the `for`-loop we can also use a plain `list` for `bucket.aux_stmts` instead of `Option<list>`.